optuna/storages/_base.py

```diff
@@ -636,6 +639,7 @@ def get_trial_system_attrs(self, trial_id: int) -> Dict[str, Any]:
         """
         return self.get_trial(trial_id).system_attrs
 
+    @abc.abstractmethod
```

Comment:
Not sure this line is needed but maybe you had something in mind? Wondering since it's a breaking change for (possibly user defined) storage classes.